src/occwl/face.py

```diff
@@ -513,29 +513,39 @@ def periodic_v(self):
         adaptor = BRepAdaptor_Surface(self.topods_shape())
         return adaptor.IsVPeriodic()
 
-    def get_triangles(self):
+    def get_triangles(self, return_normals=False):
         """
         Get the tessellation of this face as a triangle mesh
         NOTE: First you must call shape.triangulate_all_faces()
         Then call this method to get the triangles for the
         face.
 
+        Args:
+            return_normals (bool): Return vertex normals
+
         Returns:
             2D np.ndarray: Vertices
             2D np.ndarray: Faces
+            2D np.ndarray: Vertex Normals (when return_normals is True)
         """
         location = TopLoc_Location()
         bt = BRep_Tool()
         facing = bt.Triangulation(self.topods_shape(), location)
         if facing == None:
             return [], []
 
-        tab = facing.Nodes()
+        vert_nodes = facing.Nodes()
         tri = facing.Triangles()
+        uv_nodes = facing.UVNodes()
         verts = []
+        normals = []
         for i in range(1, facing.NbNodes() + 1):
-            p = tab.Value(i).Transformed(location.Transformation())
-            verts.append(np.array(list(p.Coord())))
+            vert = vert_nodes.Value(i).Transformed(location.Transformation())
+            verts.append(np.array(list(vert.Coord())))
+            if return_normals:
+                uv = uv_nodes.Value(i).Coord()
+                normal = self.normal(uv)
```

Comment:
Yes.  That looks like the right workflow.   There are some possible speedups if it's slow, but I'm sure it will be fine like this.

Reply:
Yes I was thinking one thing was to allocate the np array at the start for `verts`, `tris`, and `normals` the write directly to it. Rather than create a list.

Reply:
I was thinking of constructing the transform and [BRepAdaptor_Surface](https://dev.opencascade.org/doc/refman/html/class_b_rep_adaptor___surface.html) once and using them for each point rather than building and destroying them inside each call to `occwl.Face.normal()`

Reply:
I see. In that case we probably want a `normals()` method that takes in an array of uvs and returns all normals?

Reply:
We ideally need `points`, `normals`, `tangents`... All evaluation methods need to support multiple U-/UV-points as input.

Reply:
Might be best for a future PR when/if we hit issues.